src/interfaces/IChainlinkOracleFactory.sol

```diff
@@ -0,0 +1,36 @@
+// SPDX-License-Identifier: GPL-3.0-or-later
+pragma solidity ^0.8.17;
+
+import {ChainlinkOracleImpl} from "../chainlink/oracle/ChainlinkOracleImpl.sol";
+
+/// @title Oracle factory interface
+interface IChainlinkOracleFactory {
+    /// -----------------------------------------------------------------------
+    /// events
+    /// -----------------------------------------------------------------------
+
+    event CreateChainlinkOracle(address indexed oracle, ChainlinkOracleImpl.InitParams params);
+
+    /// -----------------------------------------------------------------------
+    /// functions - public & external
+    /// -----------------------------------------------------------------------
+
+    /**
+     * @notice Create a new Chainlink Oracle
+     * @param data_ abi encoded initial chainlink oracle params
+     * @param salt_ user defined salt
+     * @return oracle The oracle address
+     */
+    function createOracle(bytes calldata data_, bytes32 salt_) external returns (address);
+
+    /**
+     * @notice Predict the address of a new Chainlink Oracle
+     * @param data_ abi encoded initial chainlink oracle params
+     * @param salt_ user defined salt
+     * @return oracle The oracle address
+     */
+    function predictDeterministicAddress(bytes calldata data_, bytes32 salt_) external view returns (address);
```

Comment:
can we also return a bool in this fn that says whether the contract already exists at the specified address or not?

Reply:
I think thats not the purpose of this function, can have a separate function that checks if deployable or not, wdyt?

Reply:
sure, separate fn works i suppose